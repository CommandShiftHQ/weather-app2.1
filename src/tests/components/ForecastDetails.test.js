import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  it("renders correctly", () => {
    const validProps = {
      date: 1111111,
      temperature: {
        min: 12,
        max: 22,
      },
      wind: {
        speed: 5,
        direction: "n",
      },
      humidity: 50,
    };

    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
