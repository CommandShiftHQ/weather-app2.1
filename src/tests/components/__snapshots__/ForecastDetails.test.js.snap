// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <h3
      class="forecast-details__date"
    >
      Thu 1st Jan
    </h3>
    <p
      class="forecast-details__temperature"
    >
      <strong>
        Max Temperature: 
      </strong>
      22°C
    </p>
    <p
      class="forecast-details__temperature"
    >
      <strong>
        Min Temperature: 
      </strong>
      12°C
    </p>
    <p
      class="forecast-details__humidity"
    >
      <strong>
        Humidity: 
      </strong>
      50%
    </p>
    <p
      class="forecast-details__wind"
    >
      <strong>
        Humidity: 
      </strong>
      5mph
    </p>
  </div>
</DocumentFragment>
`;
